### PR TITLE
Clarify cluster state API docs

### PR DESCRIPTION
**Pull request by DaveCTurner**
The `GET _cluster/state` API is really only suitable for debugging or
diagnostics. Its response format is not documented since it changes
fairly freely between versions.

Today we mention in its docs that this API is unstable, and deliberately
omit a description of its response format, but we don't explicitly say
that it's only for diagnostics and is unsuitable for consumption by
external tools that might try and use it for monitoring.

This commit adjusts the docs to give some more explicit guidance about
how it should and shouldn't be used.

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

File: docs/reference/cluster/state.asciidoc [hunk 1]
```diff
@@ -20,44 +21,45 @@ Returns metadata about the state of the cluster.
 [[cluster-state-api-desc]]
 ==== {api-description-title}
 
-The cluster state API allows access to metadata representing the state of the
-whole cluster. This includes information such as
+The _cluster state_ is an internal data structure which keeps track of a
+variety of information needed by every node, including:
 
-* the set of nodes in the cluster
+* the identity and attributes of the other nodes in the cluster
```
Comment (jrodewig) [inline docs/reference/cluster/state.asciidoc:27]: Nit: We typically capitalize the first word of a bullet.
```suggestion
* The identity and attributes of the other nodes in the cluster
```

File: docs/reference/cluster/state.asciidoc [hunk 2]
```diff
@@ -20,44 +21,45 @@ Returns metadata about the state of the cluster.
...
 
-* all cluster-level settings
+* the cluster-wide settings
 
-* information about the indices in the cluster, including their mappings and
-  settings
+* the index metadata, including the mapping and settings for each index
```
Comment (jrodewig) [inline docs/reference/cluster/state.asciidoc:31]:
```suggestion
* Index metadata, including the mapping and settings for each index
```

File: docs/reference/cluster/state.asciidoc [hunk 3]
```diff
@@ -20,44 +21,45 @@ Returns metadata about the state of the cluster.
...
 
-* the locations of all the shards in the cluster.
+* the location and status of every shard copy in the cluster
```
Comment (jrodewig) [inline docs/reference/cluster/state.asciidoc:33]:
```suggestion
* The location and status of every shard copy in the cluster
```

File: docs/reference/cluster/state.asciidoc [hunk 4]
```diff
@@ -20,44 +21,45 @@ Returns metadata about the state of the cluster.
...
 
-NOTE: The response is an internal representation of the cluster state and its
-format may change from version to version. If possible, you should obtain any
-information from the cluster state using the other, more stable,
-<<cluster,cluster APIs>>.
+The elected master ensures that every node in the cluster has a copy of the
```
Comment (jrodewig) [inline docs/reference/cluster/state.asciidoc:35]: I'd say `master node` here for clarity.
```suggestion
The elected master node ensures that every node in the cluster has a copy of the
```

File: docs/reference/cluster/state.asciidoc [hunk 5]
```diff
@@ -20,44 +21,45 @@ Returns metadata about the state of the cluster.
...
+By default the cluster state API will route requests to the elected master node
+since this node is the authoritative source of cluster states. You can also
+retrieve the cluster state held on the node handling the API request by adding
+the query parameter `?local=true`.
 
-The cluster's `cluster_uuid` is also returned as part of the top-level response,
-in addition to the `metadata` section.
+{es} may need to expend significant effort to compute a response to this API in
+larger clusters, and the response may comprise a very large quantity of data.
+If you use this API repeatedly then your cluster may become unstable.
```
Comment (jrodewig) [inline docs/reference/cluster/state.asciidoc:47]: I'd include a comma to break the sentence up a bit.
```suggestion
If you use this API repeatedly, your cluster may become unstable.
```

File: docs/reference/cluster/state.asciidoc [hunk 6]
```diff
@@ -20,44 +21,45 @@ Returns metadata about the state of the cluster.
...
-By default, the cluster state request is routed to the master node, to ensure
-that the latest cluster state is returned. For debugging purposes, you can
-retrieve the cluster state local to a particular node by adding `local=true` to
-the query string.
+[[cluster-state-api-unstable-warning]]
+WARNING: The response is a representation of an internal data structure. Its
+format is not subject to the same compatibility guarantees as other more stable
+APIs and may change from version to version. **Do not query this API using
+external monitoring tools.** Instead, obtain the information you require using
+other, more stable, <<cluster,cluster APIs>>.
```
Comment (jrodewig) [inline docs/reference/cluster/state.asciidoc:54]:
```suggestion
other more stable <<cluster,cluster APIs>>.
```

**Review by jrodewig — APPROVED** [review]
LGTM. I left some suggestions, but they're non-blocking grammar/style nits.

**Pull request merged**